### PR TITLE
Add 'Runner.run.loop_start' hook

### PR DESCRIPTION
**Pull request by nathanielks**
Runs at the beginning of every loop iteration. Can be used to check the current status of Cavalcade, such as the number of jobs in the backlog as it passes the current database connection.

### CODE REVIEW

File: inc/class-runner.php [hunk 1]
```diff
@@ -105,6 +105,16 @@ public function run() {
 			// Check for any signals we've received
 			pcntl_signal_dispatch();
 
+			/**
+			* Action at the start of every loop iteration.
```
Comment (rmccue) [inline inc/class-runner.php]: These need to be aligned in one space so that the `*` aligns with the `/**`:

```php
/**
 * ...
 */
```

Comment (nathanielks) [reply inc/class-runner.php]: Fixed!

File: inc/class-runner.php [hunk 2]
```diff
@@ -104,6 +104,16 @@ public function run() {
...
 
+			/**
+			 * Action at the start of every loop iteration.
+			 *
+			 * PDO connection used to query against current state of Cavalcade
+			 * jobs.
+			 *
+			 * @param PDO $db PDO database connection.
+			 */
+			$this->hooks->run( 'Runner.run.loop_start', $this->db );
```
[the remark this answers is not in the document]
Comment (nathanielks) [reply inc/class-runner.php]: That's a great suggestion. I'll update that as well!

[the remark this answers is not in the document]
Comment (nathanielks) [reply inc/class-runner.php]: Updated 👍

**Pull request merged**